packages/apputils/src/instancetracker.ts

```diff
@@ -334,20 +334,39 @@ export class InstanceTracker<T extends Widget>
 
     this._restore = options;
 
-    return Promise.all(promises).then(([saved]) => {
-      return Promise.all(
-        saved.ids.map((id, index) => {
-          const value = saved.values[index];
-          const args = value && (value as any).data;
-          if (args === undefined) {
-            return state.remove(id);
-          }
-
-          // Execute the command and if it fails, delete the state restore data.
-          return registry.execute(command, args).catch(() => state.remove(id));
-        })
-      );
-    });
+    return Promise.all(promises)
```

Comment:
The `InstanceTracker#restore()` method seems like a good candidate to make `async`.

Reply:
Agreed.